tests/test_shipment.py

```diff
@@ -1,228 +1,160 @@
-    assert new_rate_id is not None
-    assert new_rate_id != rate_id
+    shipment = easypost.Shipment.create(**one_call_buy_shipment)
+    shipment.insure(amount="100")
+
+    assert shipment.insurance == "100.00"
 
 
 @pytest.mark.vcr()
-def test_smartrate(vcr):
-    to_address = {
-        "name": "Dr. Steve Brule",
-        "street1": "179 N Harbor Dr",
-        "city": "Redondo Beach",
-        "state": "CA",
-        "zip": "90277",
-        "country": "US",
-        "phone": "4153334444",
-        "email": "dr_steve_brule@gmail.com",
-    }
-    from_address = {
-        "name": "EasyPost",
-        "street1": "417 Montgomery Street",
-        "street2": "5th Floor",
-        "city": "San Francisco",
-        "state": "CA",
-        "zip": "94104",
-        "country": "US",
-        "phone": "4153334444",
-        "email": "support@easypost.com",
-    }
-    parcel = {
-        "length": 20.2,
-        "width": 10.9,
-        "height": 5,
-        "weight": 65.9,
-    }
+def test_shipment_refund(one_call_buy_shipment):
+    """Refunding a test shipment must happen within seconds of the shipment being created as test shipments naturally.
 
-    shipment = easypost.Shipment.create(
-        to_address=to_address,
-        from_address=from_address,
-        parcel=parcel,
-    )
-    assert shipment.rates
+    Follow a flow of created -> delivered to cycle through tracking events in test mode - as such anything older
+    than a few seconds in test mode may not be refundable.
+    """
+    shipment = easypost.Shipment.create(**one_call_buy_shipment)
+    shipment.refund()
+
+    assert shipment.refund_status == "submitted"
+
+
+@pytest.mark.vcr()
+def test_shipment_smartrate(basic_shipment):
+    shipment = easypost.Shipment.create(**basic_shipment)
+
+    assert shipment.rates is not None
 
     smartrates = shipment.get_smartrates()
-    assert shipment.rates[0]["id"] == smartrates[0]["id"]
+
     assert smartrates[0]["time_in_transit"]["percentile_50"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_75"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_85"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_90"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_95"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_97"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_99"] is not None
+
+
+@pytest.mark.vcr()
+def test_shipment_create_empty_objects(basic_shipment):
+    shipment_data = basic_shipment
+    shipment_data["customs_info"] = []
+    shipment_data["options"] = None
+    shipment_data["tax_identifiers"] = None
+    shipment_data["reference"] = ""
+
+    shipment = easypost.Shipment.create(**shipment_data)
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert str.startswith(shipment.id, "shp_")
+    assert shipment.options is not None
+    assert shipment.customs_info is None
```

Comment:
I apologize if I mispoke here. The test should be an empty `customs_items` nested list (see PHP here: https://github.com/EasyPost/easypost-php/blob/master/test/EasyPost/ShipmentTest.php#L237).

We should then assert against the top level `tax_identifiers` (missing) and assert the nested `customs_items`. This way, if we ever changed encoding in this lib, it'd catch the regression because we are testing all sorts of things (empty strings, empty objects - top level and nested, objects that are `None`, etc)

Reply:
My assertions are correct as well as the order, do you mean assert `tax_identifiers` is null which I don't see other libs have this yet.

Reply:
We will need to patch node.js lib as well for this one: https://github.com/EasyPost/easypost-node/blob/master/test/resources/shipment.test.js#L34

Reply:
Yes please, add the assertion requested here.

Reply:
I will create a ticket in the backlog to add the assertion in all 7 libs so I don't dismiss Nate's approval again.